src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1458,6 +1458,16 @@ void MergeTreeData::checkAlterIsPossible(const AlterCommands & commands, const S
         }
         else if (command.isRequireMutationStage(getInMemoryMetadata()))
         {
+            /// Type change for version column is allowed despite it's a part of sorting key
```

Comment:
How we ensure that type modification does not change the order?

Examples:
Int32 -> UInt32
UInt64 -> UInt32
UInt16 -> Int32

Reply:
(there are no tests that check it's prohibited)

Reply:
I don't see any problems with these changes. Yes, merges will work differently, but what can go wrong?